src/main/java/redis/clients/jedis/MultiNodePipelineBase.java

```diff
@@ -114,10 +123,13 @@ public final void sync() {
       Queue<Response<?>> queue = entry.getValue();
       Connection connection = connections.get(nodeKey);
       try {
-        List<Object> unformatted = connection.getMany(queue.size());
-        for (Object o : unformatted) {
-          queue.poll().set(o);
-        }
+        executor.submit(() -> {
+          List<Object> unformatted = connection.getMany(queue.size());
+          for (Object o : unformatted) {
+            queue.poll().set(o);
+          }
+          countDownLatch.countDown();
+        });
```

Comment:
```suggestion
      executor.submit(() -> {
        try {
          List<Object> unformatted = connection.getMany(queue.size());
          for (Object o : unformatted) {
            queue.poll().set(o);
          }
        } catch (JedisConnectionException jce) {
          log.error("Error with connection to " + nodeKey, jce);
          // cleanup the connection
          pipelinedResponsesIterator.remove();
          connections.remove(nodeKey);
          IOUtils.closeQuietly(connection);
        } finally {
          countDownLatch.countDown();
        }
      });
```
1. Need to place `countDownLatch.countDown()` in `finally` to prevent it from being broken by exceptions.
2. It is necessary to catch `RejectedExecutionException` to prevent the thread pool from rejecting tasks.

Reply:
It is crucial to execute countDown regardless of whether the above program is abnormal, otherwise the await cannot be released

Reply:
> It is crucial to execute countDown regardless of whether the above program is abnormal, otherwise the await cannot be released

So I moved it into `finally` to ensure release.